### PR TITLE
docs: convert `AccessControl.rst` to Markdown

### PR DESCRIPTION
**Pull request by MaxDesiatov**
Converts one of the doc files from RST to Markdown.

Continues work on SR-7454.

### CODE REVIEW

**Comment by MaxDesiatov** [comment]
@swift-ci please smoke test

**Comment by MaxDesiatov** [comment]
@swift-ci plase test Windows platform

**Comment by MaxDesiatov** [comment]
@swift-ci please test Windows platform

**Comment by MaxDesiatov** [comment]
@swift-ci please test Windows platform

**Comment by MaxDesiatov** [comment]
@swift-ci please test Windows platform

File: docs/AccessControl.md [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+less than or equal to the access level of the variable; this is written
+as `private(set)` or `internal(set)` before the `var` introducer.
+
+An initializer, method, subscript, or property may have any access level
+less than or equal to the access level of its type (including the
+implicit \'Self\' type), with a few additional rules:
+
+-   If a member is used to satisfy a protocol requirement, its access
+    level must be at least as high as the protocol conformance\'s; see
+    `Protocols`{.interpreted-text role="ref"} below.
```
Comment (benrimmington) [inline docs/AccessControl.md:67]:
```suggestion
    ["Protocols"](#protocols) below.
```

**Review by benrimmington — COMMENTED** [review]
[README.md][] and [contents.rst][] will also need to be updated.

[README.md]: <https://github.com/apple/swift/blob/main/docs/README.md>
[contents.rst]: <https://github.com/apple/swift/blob/main/docs/contents.rst>

**Comment by MaxDesiatov** [comment]
@swift-ci please smoke test

**Review by benrimmington — APPROVED** [review]

**Pull request merged**